src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -61,6 +61,8 @@ def _create_update_role_definition(cli_ctx, role_definition, for_update):
     else:
         role_definition = shell_safe_json_parse(role_definition)
 
+    if not isinstance(role_definition, dict):
+        raise CLIError('Invalid role defintion. A valid json object is expected')
```

Comment:
Should be `A valid dictionary JSON representation is expected.`

Reply:
sounds good